sklearn/preprocessing/data.py

```diff
@@ -1744,6 +1744,37 @@ def __init__(self, n_values="auto", categorical_features="all",
         self.sparse = sparse
         self.handle_unknown = handle_unknown
 
+    def get_feature_names(self, input_features=None):
+        """
+        Return feature names for output features
+
+        Parameters
+        ----------
+        input_features : list of string, length n_features, optional
+            String names for input features if available. By default,
+            "x0", "x1", ... "xn_features" is used.
+
+        Returns
+        -------
+        output_feature_names : list of string, length sum(self.n_values_)
+
+        """
+        check_is_fitted(self, 'n_values_')
+
+        if input_features is None:
+            input_features = ['x%d' % i for i in range(len(self.n_values_))]
+        else:
+            if len(input_features) != len(self.n_values_):
+                raise ValueError("Number of input_features must equal to "
+                                 "n_feature. it has to be of shape "
+                                 "(n_features,)")
+
+        feature_names = []
+        for (i, n_value) in enumerate(self.n_values_):
+            for j in xrange(n_value):
+                feature_names.append(input_features[i])
```

Comment:
Oh you mean adding `j` into `feature_names` to make it more clear, then the output will become something like
```python
['x0 0', 'x0 1', 'x1 0', 'x1 1', 'x1 2', 'x2 0', 'x2 1', 'x2 2', 'x2 3']
```
Am I wrong?